### DISCUSSION TITLE
How to assign output of "setup" to a variable?

### DISCUSSION THREAD

**Discussion by kgmat** [pycaret/pycaret#3329] [General]
Hello, is there a way to assign output of "setup" function to a variable?

**Comment by ngupta23** [comment]
`s = setup()`

**Reply by kgmat** [reply]
@ngupta23 : Can i get a readable information from this variable 's'. This variable is <pycaret.regression.oop.RegressionExperiment at 0x29d4b3bad40>. I am looking to convert it to a dataframe or an array.

**Reply by ngupta23** [reply]
If you are using the latest pre-release version (3.0.0rc9), you should use the OOP interface in that case. Then you can access all class internals as needed. 

Also check out the pull() functionality to get last operation data in the form of a dataframe.

**Reply by kgmat** [reply]
Thanks a lot @ngupta23. Though pull() works like a charm here. How can OOP interface be used? as you mentioned.

**Comment by kgmat** [comment]
Thanks a lot @ngupta23. Though pull() works like a charm here. How can OOP interface be used? as you mentioned.

**Reply by Yard1** [reply]
Simply call all functions as methods on the object, eg.
```python
setup()
compare_models()

# OOP:
s = setup()
s.compare_models()
```
